client.go

```diff
@@ -656,15 +684,56 @@ func (sc *SnowthClient) GetActiveNode(idsets ...[]string) *SnowthNode {
 }
 
 // DoRequest sends a request to IRONdb.
+// If the client is set to retry using other nodes on network failures, this
+// will perform those retries.
 func (sc *SnowthClient) DoRequest(node *SnowthNode,
-	method string, url string, body io.Reader, headers http.Header) (io.Reader, http.Header, error) {
-	return sc.DoRequestContext(context.Background(), node, method, url, body, headers)
+	method string, url string, body io.Reader,
+	headers http.Header) (io.Reader, http.Header, error) {
+	return sc.DoRequestContext(context.Background(), node, method, url,
+		body, headers)
 }
 
 // DoRequestContext is the context aware version of DoRequest.
+// If the client is set to retry using other nodes on network failures, this
+// will perform those retries.
 func (sc *SnowthClient) DoRequestContext(ctx context.Context, node *SnowthNode,
-	method string, url string, body io.Reader, headers http.Header) (io.Reader, http.Header, error) {
-	return sc.do(ctx, node, method, url, body, headers)
+	method string, url string, body io.Reader,
+	headers http.Header) (io.Reader, http.Header, error) {
+	retries := sc.Retries()
+	sn := node
+	for {
+		body, headers, err := sc.do(ctx, sn, method, url, body, headers)
+		if err == nil {
+			return body, headers, nil
+		}
+
+		// This is a pretty crude way to detect networking errors.
+		// But, it does work. Any broken network errors will contain dial tcp:
+		// and will be from the http.Client.
+		if !strings.Contains(err.Error(), "dial tcp:") {
```

Comment:
Would something like

```
if nerr, ok := err.(net.Error); ok && nerr.Temporary() {
```

work in this case? that should catch network specific errors dial et al.

Reply:
This does appear to work.  Thank you!!